### PR TITLE
Twip from panel must use Twitter handle not name

### PR DESCRIPTION
**Pull request by emerick**
Fixes brave/brave-browser#4597

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:

- Visit a Twitter user's page (e.g., https://twitter.com/emerick)
- Tip the user using the Brave Rewards panel (not the Brave Tip icon on the tweet)
- Complete the tip
- Press the Tweet button to tweet the tip
- Ensure that the tweet references the user's Twitter handle and not the user name (e.g., `@emerick` not `Emerick Rogul`)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/api/brave_rewards_api.cc [hunk 1]
```diff
@@ -143,6 +145,8 @@ void BraveRewardsTipTwitterUserFunction::OnTwitterPublisherInfoSaved(
 
   auto params_dict = std::make_unique<base::DictionaryValue>();
   params_dict->SetString("publisherKey", publisher_info->id);
+  params_dict->SetString(
+      "url", contents ? contents->GetLastCommittedURL().spec() : std::string());
```
Comment (emerick) [inline browser/extensions/api/brave_rewards_api.cc]: @NejcZdovc Wasn't sure if there was a better approach here... Perhaps I should have retrieved the screen name from the database, but then we'd have to expose that function in the rewards service. Wasn't sure if it was worth it, but let me know what you think. Maybe I'm missing something more obvious here.

File: components/brave_rewards/resources/tip/components/tipSite.tsx [hunk 2]
```diff
@@ -23,7 +24,12 @@ class TipSite extends React.Component<Props, {}> {
   }
 
   onTweet = () => {
-    this.actions.onTweet(this.props.publisher.name, '')
+    let name = this.props.publisher.name
+    if (this.props.publisher.provider === 'twitter') {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/tip/components/tipSite.tsx]: please check if url is null/undefined and that length is > 0

Comment (emerick) [reply components/brave_rewards/resources/tip/components/tipSite.tsx]: Fixed.

File: components/brave_rewards/resources/tip/components/tipSite.tsx [hunk 3]
```diff
@@ -23,7 +24,17 @@ class TipSite extends React.Component<Props, {}> {
...
   onTweet = () => {
-    this.actions.onTweet(this.props.publisher.name, '')
+    const url = this.props.url
+    if (!url || url.length === 0) {
+      return
+    }
+
+    let name = this.props.publisher.name
+    if (this.props.publisher.provider === 'twitter') {
+      name = `@${this.props.url.replace(/^.*\/(.*)$/, '$1')}`
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/tip/components/tipSite.tsx]: nit: `this.props.url` -> `url` as you defined it at top

File: browser/extensions/api/brave_rewards_api.cc (hunk 1 above)
Comment (NejcZdovc) [reply browser/extensions/api/brave_rewards_api.cc]: you have access to `publisher_info` object here which has url inside, so you could just do `publisher_info->url`

Comment (emerick) [reply browser/extensions/api/brave_rewards_api.cc]: Good point, fixed!

File: components/brave_rewards/resources/tip/components/tipSite.tsx (hunk 3 above)
Comment (emerick) [reply components/brave_rewards/resources/tip/components/tipSite.tsx]: Ah yes, whoops - fixed.

File: components/brave_rewards/resources/tip/components/tipSite.tsx [hunk 4]
```diff
@@ -23,7 +24,17 @@ class TipSite extends React.Component<Props, {}> {
   }
 
   onTweet = () => {
-    this.actions.onTweet(this.props.publisher.name, '')
+    const url = this.props.url
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/tip/components/tipSite.tsx]: Last thing 😃 I think that this check needs to go into provider check for twitter as some other providers maybe will not need url and will just have name

Comment (emerick) [reply components/brave_rewards/resources/tip/components/tipSite.tsx]: Makes sense, fixed.

**Pull request merged**